qt/scientific_interfaces/Direct/ALFInstrumentPresenter.cpp

```diff
@@ -20,32 +32,62 @@ ALFInstrumentPresenter::ALFInstrumentPresenter(IALFInstrumentView *view, std::un
   m_view->setUpInstrument(m_model->loadedWsName());
 }
 
-QWidget *ALFInstrumentPresenter::getLoadWidget() { return m_view->generateLoadWidget(); }
+QWidget *ALFInstrumentPresenter::getSampleLoadWidget() { return m_view->generateSampleLoadWidget(); }
+
+QWidget *ALFInstrumentPresenter::getVanadiumLoadWidget() { return m_view->generateVanadiumLoadWidget(); }
 
 ALFInstrumentWidget *ALFInstrumentPresenter::getInstrumentView() { return m_view->getInstrumentView(); }
 
 void ALFInstrumentPresenter::subscribeAnalysisPresenter(IALFAnalysisPresenter *presenter) {
   m_analysisPresenter = presenter;
 }
 
-void ALFInstrumentPresenter::loadRunNumber() {
-  auto const filepath = m_view->getFile();
+void ALFInstrumentPresenter::loadSettings() { m_view->loadSettings(); }
+
+void ALFInstrumentPresenter::saveSettings() { m_view->saveSettings(); }
+
+void ALFInstrumentPresenter::loadSample() {
+  auto const filepath = m_view->getSampleFile();
   if (!filepath) {
     return;
   }
 
   m_analysisPresenter->clear();
-  if (auto const message = loadAndTransform(*filepath)) {
-    m_view->warningBox(*message);
+
+  m_model->setSample(loadAndNormalise(*filepath));
+  m_view->setSampleRun(std::to_string(m_model->sampleRun()));
+
+  generateLoadedWorkspace();
+}
+
+void ALFInstrumentPresenter::loadVanadium() {
+  m_analysisPresenter->clear();
+
+  if (auto const filepath = m_view->getVanadiumFile()) {
+    m_model->setVanadium(loadAndNormalise(*filepath));
+    m_view->setVanadiumRun(std::to_string(m_model->vanadiumRun()));
+  } else {
+    m_model->setVanadium(nullptr);
+  }
+
+  generateLoadedWorkspace();
+}
+
+Mantid::API::MatrixWorkspace_sptr ALFInstrumentPresenter::loadAndNormalise(const std::string &pathToRun) {
+  try {
+    return m_model->loadAndNormalise(pathToRun);
+  } catch (std::exception const &ex) {
+    m_view->warningBox(ex.what());
+    return nullptr;
   }
-  m_view->setRunQuietly(std::to_string(m_model->runNumber()));
 }
 
-std::optional<std::string> ALFInstrumentPresenter::loadAndTransform(const std::string &pathToRun) {
+void ALFInstrumentPresenter::generateLoadedWorkspace() {
   try {
-    return m_model->loadAndTransform(pathToRun);
+    m_model->generateLoadedWorkspace();
   } catch (std::exception const &ex) {
-    return ex.what();
+    auto const iter = EXCEPTION_MAP.find(ex.what());
+    m_view->warningBox(iter != EXCEPTION_MAP.cend() ? iter->second : ex.what());
```

Comment:
Just a quick comment. As far as I can tell we look up by the string message and then prepend "Vanadium Normalization failed" before it? Do we need to hardcode the exception messages at the top or could we not do:

```cpp
try {
  ...
}
catch (std::exception const &ex) {
  m_view->warningBox(std::string("Vanadium normalization failed: ") + ex.what());
}
```

This way if the string from the exception is altered slightly you don't have to change anything here?

Note: I'm happy for this to be a code improvement and not a requirement for this to merge given the proximity to feature freeze.